Mypy error

Hi @NowanIlfideme 

I'm trying to use the library with mypy and have next error. Please advice

```
$ mypy --config-file tox.ini config.py
config.py:3: error: Skipping analyzing 'pydantic_yaml': found module but no type hints or library stubs
config.py:3: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
config.py:9: error: Class cannot subclass 'YamlModel' (has type 'Any')
```

Regards,
Alex

--- a/.github/workflows/python-testing.yml
+++ b/.github/workflows/python-testing.yml
@@ -37,3 +37,6 @@ jobs:
       - name: Test with pytest
         run: |
           pytest
+      - name: Test with mypy
+        run: |
+          mypy -m pydantic_yaml

--- a/pydantic_yaml/_yaml.py
+++ b/pydantic_yaml/_yaml.py
@@ -1,7 +1,9 @@
 # flake8: noqa
 
 try:
-    import ruamel.yaml as yaml
+    import ruamel.yaml as yaml  # type: ignore
+
+    # ruamel.yaml doesn't have type annotations
 
     if yaml.__version__ < "0.15.0":
         __yaml_lib__ = "ruamel-old"
@@ -9,7 +11,7 @@ try:
         __yaml_lib__ = "ruamel-new"
 except ImportError:
     try:
-        import yaml
+        import yaml  # type: ignore
 
         __yaml_lib__ = "pyyaml"
     except ImportError:

--- a/pydantic_yaml/models.py
+++ b/pydantic_yaml/models.py
@@ -3,10 +3,10 @@ from __future__ import annotations
 from pathlib import Path
 from typing import Any, Callable, Optional, Type, TypeVar, Union
 
-from pydantic.main import BaseModel
 from pydantic.error_wrappers import ErrorWrapper, ValidationError
-from pydantic.types import StrBytes
+from pydantic.main import BaseModel
 from pydantic.parse import Protocol
+from pydantic.types import StrBytes
 from pydantic.utils import ROOT_KEY
 
 from ._yaml import yaml
@@ -14,12 +14,11 @@ from ._yaml import yaml
 __all__ = ["YamlModel"]
 
 try:
-    from typing import Literal
-
-    ExtendedProto = Union[Protocol, Literal["yaml"]]
+    from typing import Literal  # type: ignore
 except ImportError:
-    # I think this would happen with Python < 3.8
-    ExtendedProto = Union[Protocol, str]
+    from typing_extensions import Literal
+
+ExtendedProto = Union[Protocol, Literal["yaml"]]
 
 
 def is_yaml_requested(content_type: str = None, proto: ExtendedProto = None):
@@ -31,7 +30,7 @@ def is_yaml_requested(content_type: str = None, proto: ExtendedProto = None):
     return is_yaml
 
 
-T = TypeVar('T', bound='YamlModel')
+T = TypeVar("T", bound="YamlModel")
 
 
 class YamlModel(BaseModel):
@@ -64,7 +63,7 @@ class YamlModel(BaseModel):
             exclude_defaults=exclude_defaults,
             exclude_none=exclude_none,
         )
-        res = yaml.safe_dump(data)
+        res = str(yaml.safe_dump(data))
         return res
 
     @classmethod
@@ -73,7 +72,7 @@ class YamlModel(BaseModel):
         b: StrBytes,
         *,
         content_type: str = None,
-        encoding: str = None,
+        encoding: str = "utf8",
         proto: ExtendedProto = None,
         allow_pickle: bool = False,
     ) -> T:

--- a/pydantic_yaml/versioned_models.py
+++ b/pydantic_yaml/versioned_models.py
@@ -5,7 +5,8 @@ from pydantic import validator
 from .models import YamlModel
 
 try:
-    from pydantic import SemVer
+    # If Pydantic implements a SemVer string (which I want to make a PR for eventually)
+    from pydantic import SemVer  # type: ignore
 except ImportError:
     from ._semver import SemVer
 

--- a/setup.cfg
+++ b/setup.cfg
@@ -43,7 +43,7 @@ test_require =
 	pytest
 
 [options.package_data]
-* = *.yaml, *.yml
+* = *.yaml, *.yml, py.typed
 
 [options.extras_require]
 pyyaml = pyyaml

--- a/setup.cfg
+++ b/setup.cfg
@@ -47,7 +47,9 @@ test_require =
 * = *.yaml, *.yml, py.typed
 
 [options.extras_require]
-pyyaml = pyyaml
+pyyaml = 
+	pyyaml
+	types-PyYAML
 ruamel = ruamel.yaml>=0.15,<0.18  # new API starting from 0.15, but old is still available
 dev = 
 	black
@@ -55,3 +57,9 @@ dev =
 	bump2version
 	pytest
 	mypy
+
+[mypy]
+warn_unused_configs = True
+
+[mypy-ruamel.*]
+ignore_missing_imports = True

--- a/setup.cfg
+++ b/setup.cfg
@@ -41,6 +41,7 @@ install_requires =
 	semver  # 2 or 3 should both work
 test_require = 
 	pytest
+	mypy
 
 [options.package_data]
 * = *.yaml, *.yml, py.typed
@@ -53,3 +54,4 @@ dev =
 	flake8
 	bump2version
 	pytest
+	mypy

